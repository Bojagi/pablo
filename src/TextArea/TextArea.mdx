---
name: TextArea
route: /textarea
menu: Components
---

import { Props, Playground } from 'docz'
import { Box } from '../Box/Box.tsx'
import { TextArea } from './TextArea.tsx'
import { IconButton } from '../IconButton/index.ts'
import { InfoText } from '../Typography/index.ts'

# TextArea

Form element that allows users to type in single line text. For multi-line text you can use the `TextArea` component.

## Examples

### Simple TextArea

The simples form of an `TextArea` is just the field itself. You can just pass a `useState` setter function to update state
as the first argument of `onChange` is the value itself. 
The second argument is the full `onChange` event of the native `text area` element.

<Playground>
  {() => {
    const [value, setValue] = React.useState('');
    return (
      <TextArea value={value} onChange={setValue} />
    );
  }}
</Playground>

### TextArea with label

A simple text area field doesn't really make sense as the context is missing. You can pass the `label` prop with the
content for the label (can be string or JSX).

<Playground>
  {() => {
    const [description, setDescription] = React.useState('');
    return (
      <TextArea label="Description" value={description} onChange={setDescription} />
    );
  }}
</Playground>

### TextArea with additional information

Additionally you can also pass the `infoText` prop to give hints on how to fill the field.

<Playground>
  {() => {
    const [name, setName] = React.useState('');
    return (
      <TextArea label="Name" infoText="Your name can be reeeeeeeaaaaaally long!" value={name} onChange={setName} />
    );
  }}
</Playground>

### TextArea with error

When you do validation on a field and the user input isn't valid, you can use the `error` prop and overwrite the `infoText` content with it.

<Playground>
  {() => {
    const [description, setDescription] = React.useState('');
    const hasError = React.useMemo(() => description.length && description.includes('\n'));
    return (
      <TextArea label="Description" error={hasError && "Do not write more than one line!"} infoText={"Write something about yourself"} value={description} onChange={setDescription} />
    );
  }}
</Playground>

### TextArea with end or start component

Sometimes you want to make the text area more interactive. For this you can use components passed to the `end` or `start` prop.

<Playground>
  {() => {
    const [tweet, setTweet] = React.useState('');
    return (
      <TextArea 
        label="Tweet" 
        value={tweet} 
        onChange={setTweet} 
        end={
          <Box minWidth={40} px={4}>
            <InfoText textColor={tweet.length > 140 ? 'negative.main' : 'text.info'}>
              {tweet.length}
            </InfoText>
          </Box>
        } 
      />
    );
  }}
</Playground>

## Import

```js
import { TextArea } from '@bojagi/pablo/TextArea';
```

## Props
<Props of={TextArea} />
