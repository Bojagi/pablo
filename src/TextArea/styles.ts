import { inputBaseStyles, InputBaseStyles } from '../shared/inputBaseStyles';

export interface TextAreaStyles extends InputBaseStyles {
  defaultRows: number;
}

export const textareaStyles: TextAreaStyles = {
  ...inputBaseStyles,
  defaultWidth: 500,
  defaultRows: 3,
};

console.log('textareaStyles', textareaStyles);
