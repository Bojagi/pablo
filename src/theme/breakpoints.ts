export type Breakpoint = 'sm' | 'md' | 'lg' | 'xl';

export const breakpointNames: Breakpoint[] = ['sm', 'md', 'lg', 'xl'];

export type BreakpointsArray = [number, number, number, number];

export type Breakpoints = BreakpointsArray &
  Record<Breakpoint, number> & {
    breakpointNames: Breakpoint[];
  };

export function createBreakpoints(
  array: BreakpointsArray,
  usedBreakpointNames: Breakpoint[] = breakpointNames
): Breakpoints {
  const newBreakpoints = ([...array] as unknown) as Breakpoints;
  usedBreakpointNames.forEach((bp, index) => {
    newBreakpoints[bp] = newBreakpoints[index];
  });
  newBreakpoints.breakpointNames = usedBreakpointNames;

  return newBreakpoints as Breakpoints;
}

export const defaultBreakpoints: BreakpointsArray = [600, 1000, 1200, 1920];

export const breakpoints = createBreakpoints(defaultBreakpoints, breakpointNames);
