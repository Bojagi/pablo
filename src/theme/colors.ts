export type ShadedColor = {
  main: string;
  dark: string;
  darkest: string;
  light: string;
  lightest: string;
  contrastText: string;
  contrastTextLight: string;
};

export type CommonColors = {
  white: string;
  black: string;
  whiteContrastText: string;
  blackContrastText: string;
};

export type TextColors = {
  main: string;
  info: string;
};

export type BorderColors = {
  main: string;
  lightest: string;
};

export interface GrayColors {
  '50': string;
  '100': string;
  '200': string;
  '300': string;
  '400': string;
  '500': string;
  '600': string;
  '700': string;
  '800': string;
  '900': string;
}

export type AllColors = ShadedColor & TextColors & BorderColors & CommonColors;

export interface Colors {
  common: CommonColors;
  gray: GrayColors;
  borders: BorderColors;
  brand: ShadedColor;
  positive: ShadedColor;
  negative: ShadedColor;
  neutral: ShadedColor;
  text: TextColors;
}

export const WHITE = '#ffffff';
export const BLACK = '#000000';

export const colors: Colors = {
  common: {
    white: WHITE,
    black: BLACK,
    whiteContrastText: BLACK,
    blackContrastText: WHITE,
  },
  gray: {
    '50': getGrayColor(0.95),
    '100': getGrayColor(0.9),
    '200': getGrayColor(0.8),
    '300': getGrayColor(0.7),
    '400': getGrayColor(0.6),
    '500': getGrayColor(0.5),
    '600': getGrayColor(0.4),
    '700': getGrayColor(0.3),
    '800': getGrayColor(0.2),
    '900': getGrayColor(0.1),
  },
  borders: {
    main: `${BLACK}${getHexByte(0.25)}`,
    lightest: `${BLACK}${getHexByte(0.1)}`,
  },
  text: {
    main: BLACK,
    info: `${BLACK}${getHexByte(0.5)}`,
  },
  brand: {
    lightest: '#EBE7FF',
    light: '#D6CEFF',
    main: '#6A50F2',
    dark: '#4129BD',
    darkest: '#281683',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  positive: {
    lightest: '#DFF6E6',
    light: '#BBECCB',
    main: '#38C765',
    dark: '#2FB85A',
    darkest: '#108B36',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  negative: {
    lightest: '#FFE7E5',
    light: '#FFCFCC',
    main: '#F21D0D',
    dark: '#CA271B',
    darkest: '#871A12',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  neutral: {
    lightest: '#FCF5DB',
    light: '#FFEB9E',
    main: '#FFDD5B',
    dark: '#EDC52A',
    darkest: '#D6AC0B',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
};

/**
 * Get hex color byte (e.g. to get transparency)
 * @param value Percentage value (between 0.0 and 1.0)
 */
function getHexByte(value: number) {
  return (Math.round((255 * value) / 16) * 16).toString(16);
}

function getGrayColor(value: number) {
  return `#${getHexByte(value)}${getHexByte(value)}${getHexByte(value)}`;
}
