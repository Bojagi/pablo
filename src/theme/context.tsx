import * as React from 'react';
import { PabloTheme, ComponentStyles } from './types';

export const pabloThemeContext = React.createContext<PabloTheme>({} as any);
export const pabloComponentStylesContext = React.createContext<ComponentStyles>({} as any);
export const rootContext = React.createContext<Document | ShadowRoot | null>(null);

export const useComponentStyleContext = () => React.useContext(pabloComponentStylesContext);

export const usePabloTheme = () => React.useContext(pabloThemeContext);
