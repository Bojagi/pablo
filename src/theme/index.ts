export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
