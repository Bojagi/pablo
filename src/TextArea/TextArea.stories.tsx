import React from 'react';
import { css } from 'styled-components';
import { Box } from '../Box';
import { TextArea, TextAreaProps } from './TextArea';

export default {
  title: 'TextArea',
};

const ControlledTextArea = ({ value: valueInitial, ...props }: Omit<TextAreaProps, 'onChange'>) => {
  const [value, setValue] = React.useState(valueInitial);
  return <TextArea rows={3} value={value} mb={4} {...props} onChange={setValue} />;
};

const baseStory = (args) => <ControlledTextArea {...args} />;

export const Inline = baseStory.bind(null);
Inline.args = {};

export const FullWidth = baseStory.bind(null);
FullWidth.args = { fullWidth: true };

export const FixedWidth = baseStory.bind(null);
FixedWidth.args = { width: 800 };

export const WithLabel = baseStory.bind(null);
WithLabel.args = {
  label: <>name</>,
};

export const WithInfoText = baseStory.bind(null);
WithInfoText.args = {
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
};

export const WithLabelAndInfoText = baseStory.bind(null);
WithLabelAndInfoText.args = {
  label: <>name</>,
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
};

export const Error = baseStory.bind(null);
Error.args = {
  error: 'Something terrible happened!',
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
};

export const MultipleTextAreas = () => (
  <Box>
    <Box mb={5}>
      <ControlledTextArea label="first name" />
    </Box>
    <Box mb={5}>
      <ControlledTextArea label="last name" />
    </Box>
    <Box mb={5}>
      <ControlledTextArea label="credit card no." infoText="optional" />
    </Box>
  </Box>
);

export const Outline = baseStory.bind(null);
Outline.args = { variant: 'outline' };

export const WithCustomStyles = baseStory.bind(null);
WithCustomStyles.args = {
  label: <>name</>,
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
  customStyles: {
    label: css`
      transform: rotate(6deg);
      transform-origin: 0 0;
    `,
    infoText: css`
      transform: rotate(-3deg);
      transform-origin: 0 0;
    `,
    wrapper: css`
      transform: rotate(-5deg);
      transform-origin: 0 0;
      background-color: blue;
    `,
    field: css`
      color: red;
    `,
  },
};
