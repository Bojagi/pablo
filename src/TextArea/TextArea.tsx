import React, { forwardRef } from 'react';
import styled from '@emotion/styled';
import { LayoutBoxProps } from '../Box';
import { getComponentStyle } from '../styleHelpers/getComponentStyle';
import { BaseInput, InnerInputProps, InputVariant } from '../shared/BaseInput';
import { useComponentStyle } from '../theme/useComponentStyle';
import { getCustomStyles } from '../utils/useCustomStyles';
import { baseStyle } from '../shared/baseStyle';

export interface TextAreaProps extends LayoutBoxProps {
  id?: string;
  value?: string;
  inputRef?: React.Ref<HTMLTextAreaElement>;
  error?: React.ReactNode;
  label?: React.ReactNode;
  variant?: InputVariant;
  infoText?: React.ReactNode;
  rows?: number;
  fullWidth?: boolean;
  onChange: (newValue: string, e: React.FormEvent<HTMLTextAreaElement>) => void;
}

type InnerTextAreaProps = InnerInputProps<{
  rows: number | undefined;
}>;

const InnerTextArea = styled.textarea<InnerTextAreaProps>`
  ${baseStyle}
  resize: none;
  border: 0;
  flex-grow: 1;
  padding: ${getComponentStyle('textArea.padding')};
  background-color: transparent;
  font-family: ${getComponentStyle('textArea.fontFamily')};
  font-size: ${getComponentStyle('textArea.fontSize')};
  outline: none;
  ${getCustomStyles('textArea.styles', 'field')}
`;

export const TextArea = forwardRef<HTMLDivElement, TextAreaProps>(
  ({ rows, variant = 'filled', width, ...props }, ref) => {
    const rowsFallback = useComponentStyle('textArea.defaultRows') as number;
    const rowsWithFallback = rows || rowsFallback;
    const defaultWidth = useComponentStyle('textArea.defaultWidth');

    return (
      <BaseInput
        variant={variant}
        name="textArea"
        innerRef={ref}
        {...props}
        width={width || defaultWidth}
        rows={rowsWithFallback}
        inputComponent={InnerTextArea}
      />
    );
  }
);
