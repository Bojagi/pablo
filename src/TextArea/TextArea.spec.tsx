import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';
import { PabloThemeProvider } from '../theme';
import { TextArea } from './TextArea';

test('Render without value', () => {
  const { container, getByTestId } = renderComponent({});
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-textarea')).toHaveValue('');
});

test('Render with value', () => {
  const { getByTestId } = renderComponent({ value: 'some' });
  expect(getByTestId('pbl-textarea')).toHaveValue('some');
});

test('Show label when available', () => {
  const { container, getByTestId } = renderComponent({
    value: 'some',
    label: (
      <>
        Your <strong>name</strong>
      </>
    ),
  });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-textarea-label')).toHaveTextContent('Your name');
});

test('Show infotext when available', () => {
  const { container, getByTestId } = renderComponent({
    value: 'some',
    infoText: (
      <>
        Something that <strong>helps</strong>
      </>
    ),
  });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-textarea-infotext')).toHaveTextContent('Something that helps');
});

test('Show error when an error message is given', () => {
  const { container, getByTestId } = renderComponent({
    value: 'some',
    error: (
      <>
        Something <strong>terrible</strong> happened!
      </>
    ),
  });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-textarea-infotext')).toHaveTextContent('Something terrible happened!');
});

test('Show error instead of infotext when both are set', () => {
  const { getByTestId } = renderComponent({
    value: 'some',
    error: (
      <>
        Something <strong>terrible</strong> happened!
      </>
    ),
    infoText: (
      <>
        something that <strong>helps</strong>
      </>
    ),
  });
  expect(getByTestId('pbl-textarea-infotext')).toHaveTextContent('Something terrible happened!');
});

test('Render with custom width when "width" prop is set', () => {
  const { getByTestId } = renderComponent({
    width: '600em',
  });

  expect(getByTestId('pbl-textarea-wrapper')).toHaveStyleRule('width', '600em');
});

test('Render with custom px width when number "width" prop is set', () => {
  const { getByTestId } = renderComponent({
    width: 600,
  });

  expect(getByTestId('pbl-textarea-wrapper')).toHaveStyleRule('width', '600px');
});

test('Render with width 100% when "fullWidth" prop is set', () => {
  const { getByTestId } = renderComponent({
    width: 600,
    fullWidth: true,
  });

  expect(getByTestId('pbl-textarea-wrapper')).toHaveStyleRule('width', '100%');
});

test('Trigger onChange event when the content of the inner textarea changed', () => {
  const onChangeMock = jest.fn();
  const { getByTestId } = renderComponent({
    value: 'some',
    onChange: onChangeMock,
  });

  expect(onChangeMock).toHaveBeenCalledTimes(0);
  act(() => {
    fireEvent.change(getByTestId('pbl-textarea'), { target: { value: 'newValue' } });
  });
  expect(onChangeMock).toHaveBeenCalledTimes(1);
  expect(onChangeMock).toHaveBeenCalledWith(
    'newValue',
    expect.objectContaining({ type: 'change' })
  );
});

test('Render with custom row count when "rows" prop is set', () => {
  const { getByTestId } = renderComponent({
    rows: 10,
  });

  expect(getByTestId('pbl-textarea')).toHaveAttribute('rows', '10');
});

test('Use generated id when no "id" prop is set', () => {
  const onChangeMock = jest.fn();
  const { getByTestId } = renderComponent({
    value: 'some',
    id: undefined,
    onChange: onChangeMock,
  });

  expect(getByTestId('pbl-textarea')).toHaveAttribute(
    'id',
    expect.stringContaining('pbl-textarea-')
  );
});

function renderComponent(props, customComponentStyles = {}) {
  return render(
    <PabloThemeProvider componentStyles={customComponentStyles}>
      <TextArea id="custom-id" {...props} />
    </PabloThemeProvider>
  );
}
