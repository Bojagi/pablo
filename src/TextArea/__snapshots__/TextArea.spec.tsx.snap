// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render without value 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #00000040;
  border-radius: 8px;
  background-color: #ffffff;
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  width: 500px;
}

.c2 {
  box-sizing: border-box;
  resize: none;
  border: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 6px 8px;
  background-color: transparent;
  font-family: "IBM Plex Sans",sans-serif;
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="pbl-textarea-wrapper"
      name="textarea"
      rows="3"
      width="500"
    >
      <textarea
        class="c2"
        data-testid="pbl-textarea"
        id="custom-id"
        rows="3"
      />
    </div>
  </div>
</div>
`;

exports[`Show error when an error message is given 1`] = `
.c0 {
  box-sizing: border-box;
}

.c3 {
  box-sizing: border-box;
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.75rem;
  line-height: 1.5em;
  margin-bottom: 0;
  margin-top: 4px;
  color: #F21D0D;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #00000040;
  border-radius: 8px;
  background-color: #ffffff;
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  border-color: #F21D0D;
  width: 500px;
}

.c1:focus {
  box-shadow: 0 0 0 3px #FFE7E5;
}

.c2 {
  box-sizing: border-box;
  resize: none;
  border: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 6px 8px;
  background-color: transparent;
  font-family: "IBM Plex Sans",sans-serif;
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="pbl-textarea-wrapper"
      name="textarea"
      rows="3"
      width="500"
    >
      <textarea
        class="c2"
        data-testid="pbl-textarea"
        id="custom-id"
        rows="3"
      >
        some
      </textarea>
    </div>
    <p
      class="c3"
      data-testid="pbl-textarea-infotext"
    >
      Something 
      <strong>
        terrible
      </strong>
       happened!
    </p>
  </div>
</div>
`;

exports[`Show infotext when available 1`] = `
.c0 {
  box-sizing: border-box;
}

.c3 {
  box-sizing: border-box;
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.75rem;
  line-height: 1.5em;
  margin-bottom: 0;
  margin-top: 4px;
  color: #00000080;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #00000040;
  border-radius: 8px;
  background-color: #ffffff;
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  width: 500px;
}

.c2 {
  box-sizing: border-box;
  resize: none;
  border: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 6px 8px;
  background-color: transparent;
  font-family: "IBM Plex Sans",sans-serif;
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="pbl-textarea-wrapper"
      name="textarea"
      rows="3"
      width="500"
    >
      <textarea
        class="c2"
        data-testid="pbl-textarea"
        id="custom-id"
        rows="3"
      >
        some
      </textarea>
    </div>
    <p
      class="c3"
      data-testid="pbl-textarea-infotext"
    >
      Something that 
      <strong>
        helps
      </strong>
    </p>
  </div>
</div>
`;

exports[`Show label when available 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  box-sizing: border-box;
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.4em;
  font-weight: 500;
  margin-bottom: 0.75em;
  margin-bottom: 6px;
}

.c2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #00000040;
  border-radius: 8px;
  background-color: #ffffff;
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  width: 500px;
}

.c3 {
  box-sizing: border-box;
  resize: none;
  border: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 6px 8px;
  background-color: transparent;
  font-family: "IBM Plex Sans",sans-serif;
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <label
      data-testid="pbl-textarea-label"
      for="custom-id"
    >
      <p
        class="c1"
      >
        Your 
        <strong>
          name
        </strong>
      </p>
    </label>
    <div
      class="c2"
      data-testid="pbl-textarea-wrapper"
      name="textarea"
      rows="3"
      width="500"
    >
      <textarea
        class="c3"
        data-testid="pbl-textarea"
        id="custom-id"
        rows="3"
      >
        some
      </textarea>
    </div>
  </div>
</div>
`;
